internal/controller/factory/builders.go

```diff
@@ -47,3 +48,42 @@ func reconcileSTS(ctx context.Context, rclient client.Client, crdName string, st
 	sts.Status = currentSts.Status
 	return rclient.Update(ctx, sts)
 }
+
+func reconcileConfigMap(ctx context.Context, rclient client.Client, crdName string, configMap *corev1.ConfigMap) error {
+	logger := log.FromContext(ctx)
+
+	currentConfigMap := &corev1.ConfigMap{}
+	err := rclient.Get(ctx, types.NamespacedName{Namespace: configMap.Namespace, Name: configMap.Name}, currentConfigMap)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.V(2).Info("creating new configMap", "cm_name", configMap.Name, "crd_object", crdName)
+			return rclient.Create(ctx, configMap)
+		}
+		return fmt.Errorf("cannot get existing configMap: %s, for crd_object: %s, err: %w", configMap.Name, crdName, err)
+	}
+	configMap.Annotations = labels.Merge(currentConfigMap.Annotations, configMap.Annotations)
+	if configMap.ResourceVersion != "" {
+		configMap.ResourceVersion = currentConfigMap.ResourceVersion
+	}
```

Comment:
According to field description:
**Populated by the system. Read-only.**

What's the goal of such assignment? The same for all auxiliary resources: services, sts.

Reply:
We always apply only newly generated resources. So for update call we need to reflect this value to new object from current one

Reply:
@sergeyshevch I got your point. However:
- the `configMap` is constructed on, lets say, previous step without `ResourceVersion`. Hence the check whether it is empty is redundant
- you're absolutely right that if we'll try to update outdated resource version it will result in an error. But it seems, that if we'll call update on the object constructed from scratch, then update will be applied to the latest version (in case the object with defined namespace/name is already exist) automagically. I've tested it on k8s v1.29.0, works like a charm without setting `ResourceVersion` manually. But I'm not sure about previous versions.

In total: it's not a blocker, if you consider it's ok to proceed without any changes - feel free to resolve the conversation. Just shared some thoughts.

Reply:
Good point. This code initially was copied from VM Operator. In my understanding it shouldn't work without copy of `ResourceVersion`

I guess for not we will stay with this implementation. I will create issue to test and revisit it later

Reply:
https://github.com/aenix-io/etcd-operator/issues/90 

FollowUp issue to research there more in-deep